### DISCUSSION TITLE
BAYES_SPAM not being added to email's

### DISCUSSION THREAD

**Discussion by amarc** [rspamd/rspamd#3756] [Q&A]
Hello all...

been looking for 48h now of online resources on how to configure "per user filtering rules" , in other words when "user1@domain1.com" marks/moves something as Spam it should be marked as such only for her/him (not for user2@)

So far I have in /etc/rspamd/statistic.conf: 

`classifier "bayes" {
  tokenizer {
    name = "osb";
  }
  cache {
  }
  servers = "127.0.0.1";
  backend = "redis";
  users_enabled = true;
  new_schema = true; # Always use new schema
  store_tokens = true; # Redefine if storing of tokens is desired
  signatures = false; # Store learn signatures
  per_user = true; # Enable per user classifier
  min_tokens = 1;
  backend = "redis";
  min_learns = 1;

  statfile {
    symbol = "BAYES_HAM";
    spam = false;
  }
  statfile {
    symbol = "BAYES_SPAM";
    spam = true;
  }
  learn_condition = 'return require("lua_bayes_learn").can_learn';`

Also tried same/similar config in local.d/ as statistic.conf and classifier-bayes.conf but had 50% of luck.

Basically, when something is in Spam and moved to Inbox it get's BAYES_HAM tag and it keeps delivering in Inbox. But otherway around, when something is in Inbox and get's moved to Spam it never get's BAYES_SPAM score and never get's delivered to Spam/Junk

In log's from above action out of interesting things I get:

`<d7c8cd>; csession; rspamd_controller_learn_fin_task: <127.0.0.1> learned message as spam: e38eb5f5ad4d3515b836b0cfc87a3b6f@domain.com`

but also:

`bayes; bayes_classify: no tokens found in bayes database (50 total tokens, 2 text tokens), ignore stats`

which honestly I have no idea what it means.

Where am I making misstake(s) ? Any tips would be appreciated.

ty.
